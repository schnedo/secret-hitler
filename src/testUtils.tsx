import type { PropsWithChildren } from "react";

export function MockComponent({
  children,
  ...props
}: PropsWithChildren<unknown>): JSX.Element {
  return (
    <div>
      {JSON.stringify(props)} {children}
    </div>
  );
}
