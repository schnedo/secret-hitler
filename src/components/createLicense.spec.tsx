import { render } from "@testing-library/react";
import { MockComponent } from "../testUtils";
import createLicense from "./createLicense";

const License = createLicense(MockComponent);

it("should render correctly", async () => {
  expect.hasAssertions();

  const { container } = render(
    <License open={true} onClose={() => undefined} />,
  );

  expect(container).toMatchSnapshot();
});
