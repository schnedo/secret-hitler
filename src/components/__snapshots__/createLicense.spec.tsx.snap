// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div>
    {"withBackdrop":true,"open":true}
     
    <div>
      Secret Hitler is a game developed by Goat, Wolf, & Cabbage, licensed under the
       
      <a
        href="https://creativecommons.org/licenses/by-nc-sa/4.0/deed.en"
      >
        CC BY-NC-SA 4.0
      </a>
    </div>
  </div>
</div>
`;
